make.bat

```diff
@@ -13,39 +10,24 @@ IF %errorlevel%==0 set GENERATOR=Visual Studio 16
 reg query HKEY_CLASSES_ROOT\VisualStudio.DTE.17.0 >nul 2>nul
 IF %errorlevel%==0 set GENERATOR=Visual Studio 17
 
-for /f %%i in ('git describe --tags') do set LUVI_TAG=%%i
+if "%ARCH%" == "i686" set PLATFORM=-G%GENERATOR% -AWin32
+if "%ARCH%" == "x86_64" set PLATFORM=-G%GENERATOR% -Ax64
+
+if "%PLATFORM%-%PROCESSOR_ARCHITECTURE%" == "-AMD64" set PLATFORM=-G%GENERATOR% -Ax64
+if "%PLATFORM%-" == "-" set PLATFORM=-G%GENERATOR% -AWin32
+
 IF NOT "x%1" == "x" GOTO :%1
 
 GOTO :build
 
 :regular
-ECHO "Building regular64"
-cmake -DWithOpenSSL=ON -DWithSharedOpenSSL=OFF -DWithPCRE=ON -DWithLPEG=ON -DWithSharedPCRE=OFF -H. -Bbuild -G"%GENERATOR%" -Ax64
-GOTO :end
-
-:regular-asm
-ECHO "Building regular64 asm"
-cmake -DWithOpenSSLASM=ON -DWithOpenSSL=ON -DWithSharedOpenSSL=OFF -DWithPCRE=ON -DWithLPEG=ON -DWithSharedPCRE=OFF -H. -Bbuild -G"%GENERATOR%" -Ax64
-GOTO :end
-
-:regular32
-ECHO "Building regular32"
-cmake -DWithOpenSSL=ON -DWithSharedOpenSSL=OFF -DWithPCRE=ON -DWithLPEG=ON -DWithSharedPCRE=OFF -H. -Bbuild -G"%GENERATOR%" -AWin32
-GOTO :end
-
-:regular32-asm
-ECHO "Building regular32 asm"
-cmake -DWithOpenSSLASM=ON -DWithOpenSSL=ON -DWithSharedOpenSSL=OFF -DWithPCRE=ON -DWithLPEG=ON -DWithSharedPCRE=OFF -H. -Bbuild -G"%GENERATOR%" -AWin32
+ECHO "Building regular"
+cmake -DWithOpenSSL=ON -DWithPCRE=ON -DWithLPEG=ON -H. -Bbuild "%GENERATOR%" %EXTRA_OPTIONS%
```

Comment:
This `"%GENERATOR%` seems incorrect unless I'm missing something (haven't tested it, just eyeballing it), as `GENERATOR` is just set to something like `Visual Studio 17`. Should it be `%PLATFORM%` instead?

Reply:
Ah yes it should, I didn't have a windows machine to test it on.